src/rod_mechanics/soft_robots.cpp

```diff
@@ -61,8 +61,8 @@ void SoftRobots::applyPositionBC(const MatXN<5>& delta_pos) {
     // delta_pos: the first col is limb_idx, second col is node_idx, third col
     // is dx, fourth col is dy, fifth col is dz
     for (int i = 0; i < delta_pos.rows(); i++) {
-        int limb_idx = delta_pos(i, 0);
-        int node_idx = delta_pos(i, 1);
+        int limb_idx = (int)delta_pos(i, 0);
+        int node_idx = (int)delta_pos(i, 1);
```

Comment:
Do we really need this? I think it will convert to int directly

Reply:
I think it's generally better to explicitly convert rather than implicitly convert when possible. It's clearer for readers.